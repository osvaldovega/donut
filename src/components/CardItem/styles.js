import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    width: 100,
    height: 100,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  label: {
    color: '#000',
  },
});

export default styles;
