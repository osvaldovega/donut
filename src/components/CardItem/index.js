import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';

const CardItem = ({ id, display_name, base_currency, onPressFn }) => (
  <View style={styles.wrapper} onStartShouldSetResponder={onPressFn}>
    <View style={styles.titleContainer}>
      <Text style={styles.title}>{display_name}</Text>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <Text style={styles.label}>ID</Text>
        <Text style={styles.content}>{id}</Text>
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <Text style={styles.label}>Currency</Text>
        <Text style={styles.content}>{base_currency}</Text>
      </View>
    </View>
  </View>
);

CardItem.defaultProps = {
  id: 0,
  display_name: '',
  base_currency: '',
  onPressFn: () => {},
}

export default CardItem;
