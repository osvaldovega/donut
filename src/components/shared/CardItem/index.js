import React from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import TextHeader from '../TextHeader';
import TextLabel from '../TextLabel';
import TextContent from '../TextContent';
import Button from '../Button';
import styles from './styles';

const CardItem = ({
  id,
  base_currency,
  quote_currency,
  base_min_size,
  base_max_size,
  min_market_funds,
  max_market_funds,
  quote_increment,
  status,
  onPressFn,
}) => (
  <View style={styles.wrapper}>
    <View style={styles.titleContainer}>
      <TextHeader content={id} />
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Base'} />
        <TextContent content={base_currency} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Quote'} />
        <TextContent content={quote_currency} />
      </View>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Base Order Min'} />
        <TextContent content={base_min_size} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Base Order Max'} />
        <TextContent content={base_max_size} />
      </View>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Quote Order Min'} />
        <TextContent content={min_market_funds} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Quote Order Max'} />
        <TextContent content={max_market_funds} />
      </View>
    </View>

    <View style={styles.detailsContainer}>
      <View style={styles.detailsColumn}>
        <TextLabel content={'Tik Size'} />
        <TextContent content={quote_increment} />
      </View>
      <View style={[styles.detailsColumn, styles.columnDivider]}>
        <TextLabel content={'Status'} />
        <TextContent content={status} />
      </View>
    </View>

    <Button text={'24hr Stats'} onPressFn={onPressFn}/>
  </View>
);

CardItem.defaultProps = {
  base_currency: '',
  base_max_size: 0,
  base_min_size: 0,
  id: 0,
  max_market_funds: 0,
  min_market_funds: 0,
  onPressFn: () => {},
  quote_currency: '',
  quote_increment: 0,
  status: '',
}

export default CardItem;
