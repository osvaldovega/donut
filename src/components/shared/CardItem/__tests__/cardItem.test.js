import React from 'react';
import renderer from 'react-test-renderer';
import CardItem from '../index';

describe('<CardItem />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<CardItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
