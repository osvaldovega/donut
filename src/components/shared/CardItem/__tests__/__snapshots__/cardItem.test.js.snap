// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#192a56",
      "borderColor": "#ffffff",
      "borderRadius": 6,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "flex-start",
      "marginTop": 10,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
            "width": "100%",
          },
          "",
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#44bd32",
              "fontFamily": "montserrat",
              "fontSize": 18,
              "fontWeight": "600",
            },
            "",
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Base
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Quote
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Base Order Min
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Base Order Max
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Quote Order Min
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Quote Order Max
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#f1f2f6",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Tik Size
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "50%",
          },
          Object {
            "borderLeftColor": "#f1f2f6",
            "borderLeftWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eccc68",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "textTransform": "uppercase",
              },
              "",
            ]
          }
        >
          Status
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            },
            "",
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f1f2f6",
                "fontFamily": "montserrat",
                "fontSize": 12,
                "fontWeight": "600",
                "marginTop": 5,
              },
              "",
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "auto",
        "justifyContent": "center",
        "marginBottom": 15,
        "marginTop": 15,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#1e90ff",
            "borderRadius": 9,
            "color": "#f1f2f6",
            "fontFamily": "montserrat",
            "fontSize": 18,
            "fontWeight": "600",
            "overflow": "hidden",
            "padding": 5,
            "textAlign": "center",
            "width": 200,
          },
          "",
        ]
      }
    >
      24hr Stats
    </Text>
  </View>
</View>
`;
